tozti/auth/utils.py

```diff
@@ -0,0 +1,51 @@
+import json
+from pymacaroons import Macaroon, Verifier
+import tozti
+from tozti.utils import APIError
+
+def create_macaroon(*args, **kwargs):
+    mac = Macaroon(
+        location=tozti.CONFIG['http']['host'],
+        identifier=tozti.CONFIG['cookie']['public_key'],
+        key=tozti.CONFIG['cookie']['private_key']
+    )
+
+    for a in args:
+        mac.add_first_party_caveat(json.dumps(a))
+            
+    for key, value in kwargs.items():
+        mac.add_first_party_caveat('{}={}'.format(json.dumps(key), json.dumps(value)))
+    
+    return mac
+
+class BadPasswordError(tozti.utils.APIError):
+    def __init__(self):
+        super().__init__(msg=self.detail)
```

Comment:
I'm not so fond of that, what did you want to achieve? tozti.utils.APIError should already provide a sufficient constructor. Maybe you want to change APIError to something cleaner (i would actually change the constructor to [init attributes...] super().__init__(self.to_response())). If you do that make sure it doesn't break tozti.app.error_handler.